test/utils/session-core.test.js

```diff
@@ -0,0 +1,282 @@
+import {
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  createSolidDataset,
+  setThing,
+  getSolidDataset
+} from '@inrupt/solid-client';
+import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
+import { getDocuments } from '../../src/utils/network/session-core';
+
+const mockPodUrl = 'https://pod.example.com/';
+const mockSolidIdentityProvider = 'https://example.com/';
+let session = {};
+
+describe('setDocAclPermission', () => {
+  const setDocAclForUser = vi.fn(() => Promise.resolve());
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('prints out correct documentUrl and webId for other user and calls setDocAclForUser', async () => {
```

Comment:
Unfortunately this test still isn't actually testing anything. All the `expect` statements only check data generated in the test. The `expect` on line 42 is checking the `documentUrl` value set on line 34. The `expect` on line 43 is checking the `webId` value from line 35, and the `expect` on line 41 is checking that the function call you made on line 39 has the same parameters as what you passed in on line 39.

I'd expect the test to look something like this:

```
const result = await setDocAclForUser(params);

expect(saveAclFor).toBeCalledWith(testParams); // Whatever you expect to be passed to Solid
expect(result).toMatchObject(object); // Whatever you expect the return value to look like, if there is any

Reply:
I'll try to update this to call the actual setDocAclForUser, but it doesn't return anything, so I just expect the saveAclFor to be called?